From ee41a29c3a376bb8396edf9698d13f9695a3df5e Mon Sep 17 00:00:00 2001
From: Srinivas Neeli <srinivas.neeli@xilinx.com>
Date: Mon, 23 Mar 2020 23:23:01 +0530
Subject: [PATCH 1291/1852] pinctrl: zynqmp: use module_platform_driver to
 simplify the code

commit 831e244b52414f0e31a7c34d077055250c27cc72 from
https://github.com/Xilinx/linux-xlnx.git

module_platform_driver() makes the code simpler by eliminating
boilerplate code.

Signed-off-by: Srinivas Neeli <srinivas.neeli@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/pinctrl/pinctrl-zynqmp.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-zynqmp.c b/drivers/pinctrl/pinctrl-zynqmp.c
index 9b3b4dc88a1f..c76cb6b5ad3f 100644
--- a/drivers/pinctrl/pinctrl-zynqmp.c
+++ b/drivers/pinctrl/pinctrl-zynqmp.c
@@ -9,6 +9,7 @@
  *  Chirag Parekh <chirag.parekh@xilinx.com>
  */
 
+#include <linux/module.h>
 #include <dt-bindings/pinctrl/pinctrl-zynqmp.h>
 #include <linux/firmware/xlnx-zynqmp.h>
 #include <linux/init.h>
@@ -1067,8 +1068,4 @@ static struct platform_driver zynqmp_pinctrl_driver = {
 	.remove = zynqmp_pinctrl_remove,
 };
 
-static int __init zynqmp_pinctrl_init(void)
-{
-	return platform_driver_register(&zynqmp_pinctrl_driver);
-}
-arch_initcall(zynqmp_pinctrl_init);
+module_platform_driver(zynqmp_pinctrl_driver);
-- 
2.31.1

